{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c3fc502b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0546dc93",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "DATA_DIR = \"../data/ccle/ccle_broad_2019\"\n",
    "EXPR_FILE = f\"{DATA_DIR}/data_mrna_seq_rpkm.txt\"\n",
    "META_EXPR_FILE = f\"{DATA_DIR}/meta_mrna_seq_rpkm.txt\"\n",
    "META_STUDY_FILE = f\"{DATA_DIR}/meta_study.txt\"\n",
    "MUTATIONS_FILE=f\"{DATA_DIR}/data_mutations.txt\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "c90c5670",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # 3. Load Data\n",
    "# Read the expression matrix and metadata into pandas DataFrames.\n",
    "\n",
    "# Expression data (genes x cell lines)\n",
    "expr = pd.read_csv(EXPR_FILE, sep='\\t', index_col=0)\n",
    "# Metadata for expression samples\n",
    "meta_expr = pd.read_csv(META_EXPR_FILE, sep='\\t', index_col=0)\n",
    "# Study metadata (cell line annotations)\n",
    "meta_study = pd.read_csv(META_STUDY_FILE, sep='\\t', index_col=0)\n",
    "\n",
    "#mutations file\n",
    "mutations = pd.read_csv(MUTATIONS_FILE,  sep='\\t', comment = '#' ,low_memory=False, index_col=0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7444f51b",
   "metadata": {},
   "outputs": [],
   "source": [
    "def select_mutation_columns(file_path):\n",
    "    \"\"\"\n",
    "    Load mutations data and select only Hugo_Symbol, Tumor_Sample_Barcode, \n",
    "    Variant_Type, and Variant_Classification columns\n",
    "    \"\"\"\n",
    "    # Read the full dataset\n",
    "    df = pd.read_csv(file_path, sep='\\t', comment='#', low_memory=False)\n",
    "    \n",
    "    # Select only the columns you need\n",
    "    columns_needed = [\n",
    "        'Hugo_Symbol',           # Gene symbol\n",
    "        'Tumor_Sample_Barcode',  # Sample name\n",
    "        'Variant_Type',          # Variant type (SNP, INS, DEL, etc.)\n",
    "        'Variant_Classification' # Variant classification (Missense, Silent, etc.)\n",
    "    ]\n",
    "    \n",
    "    # Create subset with only these columns\n",
    "    df_subset = df[columns_needed].copy()\n",
    "    \n",
    "    return df_subset\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "86f4b743",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>DDX11L1</th>\n",
       "      <th>WASH7P</th>\n",
       "      <th>MIR1302-11</th>\n",
       "      <th>FAM138A</th>\n",
       "      <th>OR4G4P</th>\n",
       "      <th>OR4G11P</th>\n",
       "      <th>OR4F5</th>\n",
       "      <th>RP11-34P13.7</th>\n",
       "      <th>CICP27</th>\n",
       "      <th>AL627309.1</th>\n",
       "      <th>...</th>\n",
       "      <th>MT-ND3</th>\n",
       "      <th>MT-TR</th>\n",
       "      <th>MT-TH</th>\n",
       "      <th>MT-TS2</th>\n",
       "      <th>MT-TL2</th>\n",
       "      <th>MT-ND5</th>\n",
       "      <th>MT-ND6</th>\n",
       "      <th>MT-TE</th>\n",
       "      <th>MT-CYB</th>\n",
       "      <th>MT-TT</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>22RV1_PROSTATE</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>12.63011</td>\n",
       "      <td>0.04289</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.44714</td>\n",
       "      <td>0.37909</td>\n",
       "      <td>6.17780</td>\n",
       "      <td>...</td>\n",
       "      <td>1176.81189</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.12335</td>\n",
       "      <td>2393.80298</td>\n",
       "      <td>1825.52332</td>\n",
       "      <td>0.25386</td>\n",
       "      <td>4923.78223</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2313287_STOMACH</th>\n",
       "      <td>0.03755</td>\n",
       "      <td>10.14155</td>\n",
       "      <td>0.01037</td>\n",
       "      <td>0.00869</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.34327</td>\n",
       "      <td>0.09168</td>\n",
       "      <td>6.56648</td>\n",
       "      <td>...</td>\n",
       "      <td>1009.34979</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1960.96802</td>\n",
       "      <td>2056.64697</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>3827.00537</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253JBV_URINARY_TRACT</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>6.20657</td>\n",
       "      <td>0.16955</td>\n",
       "      <td>0.22423</td>\n",
       "      <td>0.02886</td>\n",
       "      <td>0.01939</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.16982</td>\n",
       "      <td>0.10995</td>\n",
       "      <td>1.03405</td>\n",
       "      <td>...</td>\n",
       "      <td>2816.78296</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.39614</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.51331</td>\n",
       "      <td>1896.28870</td>\n",
       "      <td>2184.88647</td>\n",
       "      <td>0.26410</td>\n",
       "      <td>1905.18481</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>253J_URINARY_TRACT</th>\n",
       "      <td>0.06507</td>\n",
       "      <td>6.55835</td>\n",
       "      <td>0.23364</td>\n",
       "      <td>0.16558</td>\n",
       "      <td>0.03875</td>\n",
       "      <td>0.03904</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.08922</td>\n",
       "      <td>0.09146</td>\n",
       "      <td>0.76888</td>\n",
       "      <td>...</td>\n",
       "      <td>7116.30469</td>\n",
       "      <td>0.14115</td>\n",
       "      <td>0.26594</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1.03378</td>\n",
       "      <td>3522.37695</td>\n",
       "      <td>4568.00195</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>4590.01416</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42MGBA_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>7.61752</td>\n",
       "      <td>0.08765</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.09428</td>\n",
       "      <td>0.07513</td>\n",
       "      <td>1.01231</td>\n",
       "      <td>...</td>\n",
       "      <td>1049.68384</td>\n",
       "      <td>0.13768</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.25209</td>\n",
       "      <td>1178.93579</td>\n",
       "      <td>1616.96484</td>\n",
       "      <td>0.12970</td>\n",
       "      <td>2343.92407</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UMUC9_URINARY_TRACT</th>\n",
       "      <td>0.03236</td>\n",
       "      <td>8.70551</td>\n",
       "      <td>0.07747</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00642</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01325</td>\n",
       "      <td>0.02958</td>\n",
       "      <td>0.10214</td>\n",
       "      <td>0.44584</td>\n",
       "      <td>...</td>\n",
       "      <td>2820.28882</td>\n",
       "      <td>0.09360</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1452.41895</td>\n",
       "      <td>996.53717</td>\n",
       "      <td>0.08817</td>\n",
       "      <td>3437.12769</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UPCISCC152_UPPER_AERODIGESTIVE_TRACT</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>15.23348</td>\n",
       "      <td>0.07448</td>\n",
       "      <td>0.02674</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.04402</td>\n",
       "      <td>0.08265</td>\n",
       "      <td>1.91472</td>\n",
       "      <td>...</td>\n",
       "      <td>995.56390</td>\n",
       "      <td>0.16713</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.30602</td>\n",
       "      <td>793.58710</td>\n",
       "      <td>593.61139</td>\n",
       "      <td>0.31489</td>\n",
       "      <td>1645.22168</td>\n",
       "      <td>0.00000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UW228_CENTRAL_NERVOUS_SYSTEM</th>\n",
       "      <td>0.03799</td>\n",
       "      <td>6.47732</td>\n",
       "      <td>0.01049</td>\n",
       "      <td>0.00879</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.02279</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.06945</td>\n",
       "      <td>0.01686</td>\n",
       "      <td>0.21741</td>\n",
       "      <td>...</td>\n",
       "      <td>441.25647</td>\n",
       "      <td>0.16482</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>757.41357</td>\n",
       "      <td>472.01627</td>\n",
       "      <td>0.15526</td>\n",
       "      <td>973.56439</td>\n",
       "      <td>0.16232</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Y79_AUTONOMIC_GANGLIA</th>\n",
       "      <td>0.02907</td>\n",
       "      <td>8.45568</td>\n",
       "      <td>0.39607</td>\n",
       "      <td>0.26449</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.01163</td>\n",
       "      <td>0.01191</td>\n",
       "      <td>0.07971</td>\n",
       "      <td>0.09748</td>\n",
       "      <td>1.05557</td>\n",
       "      <td>...</td>\n",
       "      <td>2341.42505</td>\n",
       "      <td>0.50444</td>\n",
       "      <td>0.23760</td>\n",
       "      <td>0.27787</td>\n",
       "      <td>0.92361</td>\n",
       "      <td>1218.16992</td>\n",
       "      <td>729.65912</td>\n",
       "      <td>0.55439</td>\n",
       "      <td>1221.39954</td>\n",
       "      <td>0.08280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>YAMATO_SOFT_TISSUE</th>\n",
       "      <td>0.00000</td>\n",
       "      <td>11.81688</td>\n",
       "      <td>0.17586</td>\n",
       "      <td>0.02266</td>\n",
       "      <td>0.01459</td>\n",
       "      <td>0.01469</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.05597</td>\n",
       "      <td>0.36233</td>\n",
       "      <td>1.28208</td>\n",
       "      <td>...</td>\n",
       "      <td>904.73688</td>\n",
       "      <td>0.21249</td>\n",
       "      <td>0.20018</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1320.84558</td>\n",
       "      <td>564.16779</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>1755.82764</td>\n",
       "      <td>0.20928</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1156 rows × 56318 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol                           DDX11L1    WASH7P  MIR1302-11  FAM138A  \\\n",
       "Tumor_Sample_Barcode                                                           \n",
       "22RV1_PROSTATE                        0.00000  12.63011     0.04289  0.00000   \n",
       "2313287_STOMACH                       0.03755  10.14155     0.01037  0.00869   \n",
       "253JBV_URINARY_TRACT                  0.00000   6.20657     0.16955  0.22423   \n",
       "253J_URINARY_TRACT                    0.06507   6.55835     0.23364  0.16558   \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         0.00000   7.61752     0.08765  0.00000   \n",
       "...                                       ...       ...         ...      ...   \n",
       "UMUC9_URINARY_TRACT                   0.03236   8.70551     0.07747  0.00000   \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT  0.00000  15.23348     0.07448  0.02674   \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM          0.03799   6.47732     0.01049  0.00879   \n",
       "Y79_AUTONOMIC_GANGLIA                 0.02907   8.45568     0.39607  0.26449   \n",
       "YAMATO_SOFT_TISSUE                    0.00000  11.81688     0.17586  0.02266   \n",
       "\n",
       "Hugo_Symbol                            OR4G4P  OR4G11P    OR4F5  RP11-34P13.7  \\\n",
       "Tumor_Sample_Barcode                                                            \n",
       "22RV1_PROSTATE                        0.00000  0.00000  0.00000       0.44714   \n",
       "2313287_STOMACH                       0.00000  0.00000  0.00000       0.34327   \n",
       "253JBV_URINARY_TRACT                  0.02886  0.01939  0.00000       0.16982   \n",
       "253J_URINARY_TRACT                    0.03875  0.03904  0.00000       0.08922   \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         0.00000  0.00000  0.00000       0.09428   \n",
       "...                                       ...      ...      ...           ...   \n",
       "UMUC9_URINARY_TRACT                   0.00642  0.00000  0.01325       0.02958   \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT  0.00000  0.00000  0.00000       0.04402   \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM          0.00000  0.02279  0.00000       0.06945   \n",
       "Y79_AUTONOMIC_GANGLIA                 0.00000  0.01163  0.01191       0.07971   \n",
       "YAMATO_SOFT_TISSUE                    0.01459  0.01469  0.00000       0.05597   \n",
       "\n",
       "Hugo_Symbol                            CICP27  AL627309.1  ...      MT-ND3  \\\n",
       "Tumor_Sample_Barcode                                       ...               \n",
       "22RV1_PROSTATE                        0.37909     6.17780  ...  1176.81189   \n",
       "2313287_STOMACH                       0.09168     6.56648  ...  1009.34979   \n",
       "253JBV_URINARY_TRACT                  0.10995     1.03405  ...  2816.78296   \n",
       "253J_URINARY_TRACT                    0.09146     0.76888  ...  7116.30469   \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         0.07513     1.01231  ...  1049.68384   \n",
       "...                                       ...         ...  ...         ...   \n",
       "UMUC9_URINARY_TRACT                   0.10214     0.44584  ...  2820.28882   \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT  0.08265     1.91472  ...   995.56390   \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM          0.01686     0.21741  ...   441.25647   \n",
       "Y79_AUTONOMIC_GANGLIA                 0.09748     1.05557  ...  2341.42505   \n",
       "YAMATO_SOFT_TISSUE                    0.36233     1.28208  ...   904.73688   \n",
       "\n",
       "Hugo_Symbol                             MT-TR    MT-TH   MT-TS2   MT-TL2  \\\n",
       "Tumor_Sample_Barcode                                                       \n",
       "22RV1_PROSTATE                        0.00000  0.00000  0.00000  0.12335   \n",
       "2313287_STOMACH                       0.00000  0.00000  0.00000  0.00000   \n",
       "253JBV_URINARY_TRACT                  0.00000  0.39614  0.00000  0.51331   \n",
       "253J_URINARY_TRACT                    0.14115  0.26594  0.00000  1.03378   \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         0.13768  0.00000  0.00000  0.25209   \n",
       "...                                       ...      ...      ...      ...   \n",
       "UMUC9_URINARY_TRACT                   0.09360  0.00000  0.00000  0.00000   \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT  0.16713  0.00000  0.00000  0.30602   \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM          0.16482  0.00000  0.00000  0.00000   \n",
       "Y79_AUTONOMIC_GANGLIA                 0.50444  0.23760  0.27787  0.92361   \n",
       "YAMATO_SOFT_TISSUE                    0.21249  0.20018  0.00000  0.00000   \n",
       "\n",
       "Hugo_Symbol                               MT-ND5      MT-ND6    MT-TE  \\\n",
       "Tumor_Sample_Barcode                                                    \n",
       "22RV1_PROSTATE                        2393.80298  1825.52332  0.25386   \n",
       "2313287_STOMACH                       1960.96802  2056.64697  0.00000   \n",
       "253JBV_URINARY_TRACT                  1896.28870  2184.88647  0.26410   \n",
       "253J_URINARY_TRACT                    3522.37695  4568.00195  0.00000   \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         1178.93579  1616.96484  0.12970   \n",
       "...                                          ...         ...      ...   \n",
       "UMUC9_URINARY_TRACT                   1452.41895   996.53717  0.08817   \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT   793.58710   593.61139  0.31489   \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM           757.41357   472.01627  0.15526   \n",
       "Y79_AUTONOMIC_GANGLIA                 1218.16992   729.65912  0.55439   \n",
       "YAMATO_SOFT_TISSUE                    1320.84558   564.16779  0.00000   \n",
       "\n",
       "Hugo_Symbol                               MT-CYB    MT-TT  \n",
       "Tumor_Sample_Barcode                                       \n",
       "22RV1_PROSTATE                        4923.78223  0.00000  \n",
       "2313287_STOMACH                       3827.00537  0.00000  \n",
       "253JBV_URINARY_TRACT                  1905.18481  0.00000  \n",
       "253J_URINARY_TRACT                    4590.01416  0.00000  \n",
       "42MGBA_CENTRAL_NERVOUS_SYSTEM         2343.92407  0.00000  \n",
       "...                                          ...      ...  \n",
       "UMUC9_URINARY_TRACT                   3437.12769  0.00000  \n",
       "UPCISCC152_UPPER_AERODIGESTIVE_TRACT  1645.22168  0.00000  \n",
       "UW228_CENTRAL_NERVOUS_SYSTEM           973.56439  0.16232  \n",
       "Y79_AUTONOMIC_GANGLIA                 1221.39954  0.08280  \n",
       "YAMATO_SOFT_TISSUE                    1755.82764  0.20928  \n",
       "\n",
       "[1156 rows x 56318 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expr = expr.transpose()\n",
    "expr.index.name = 'Tumor_Sample_Barcode'\n",
    "expr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "3817b56c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_AC</th>\n",
       "      <th>SangerRecalibWES_AC</th>\n",
       "      <th>SangerWES_AC</th>\n",
       "      <th>TCGAhsCnt</th>\n",
       "      <th>WES_AC</th>\n",
       "      <th>WGS_AC</th>\n",
       "      <th>cDNA_Change</th>\n",
       "      <th>isCOSMIChotspot</th>\n",
       "      <th>isDeleterious</th>\n",
       "      <th>isTCGAhotspot</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>DVL1</th>\n",
       "      <td>1855</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>1277461</td>\n",
       "      <td>1277461</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>87:39</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.438G&gt;A</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AL590822.1</th>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>2144416</td>\n",
       "      <td>2144416</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>23:43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.604C&gt;T</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLCH2</th>\n",
       "      <td>9651</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>2435359</td>\n",
       "      <td>2435359</td>\n",
       "      <td>+</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>3'UTR</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>9:25</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UBE4B</th>\n",
       "      <td>10277</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>10177641</td>\n",
       "      <td>10177641</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>13:95</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.934G&gt;A</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SRM</th>\n",
       "      <td>6723</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>11119363</td>\n",
       "      <td>11119363</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>113:57</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.207C&gt;A</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MLLT4</th>\n",
       "      <td>4301</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>168366687</td>\n",
       "      <td>168366689</td>\n",
       "      <td>+</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>3'Flank</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>12:91</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.5198_5200delACA</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-ND2</th>\n",
       "      <td>4536</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37</td>\n",
       "      <td>M</td>\n",
       "      <td>4604</td>\n",
       "      <td>4605</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>9:18</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.135_136insA</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>POLR3H</th>\n",
       "      <td>171568</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>22</td>\n",
       "      <td>41940458</td>\n",
       "      <td>41940459</td>\n",
       "      <td>+</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>5'Flank</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>5:3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-ND5</th>\n",
       "      <td>4540</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37</td>\n",
       "      <td>M</td>\n",
       "      <td>12384</td>\n",
       "      <td>12385</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>12:69</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.48_49insC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MT-CO3</th>\n",
       "      <td>4514</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37</td>\n",
       "      <td>M</td>\n",
       "      <td>9531</td>\n",
       "      <td>9532</td>\n",
       "      <td>+</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>10:29</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.325_326insC</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1212258 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Entrez_Gene_Id  Center NCBI_Build Chromosome  Start_Position  \\\n",
       "Hugo_Symbol                                                                 \n",
       "DVL1                   1855     NaN     GRCh37          1         1277461   \n",
       "AL590822.1                0     NaN     GRCh37          1         2144416   \n",
       "PLCH2                  9651     NaN     GRCh37          1         2435359   \n",
       "UBE4B                 10277     NaN     GRCh37          1        10177641   \n",
       "SRM                    6723     NaN     GRCh37          1        11119363   \n",
       "...                     ...     ...        ...        ...             ...   \n",
       "MLLT4                  4301     NaN     GRCh37          6       168366687   \n",
       "MT-ND2                 4536     NaN         37          M            4604   \n",
       "POLR3H               171568     NaN     GRCh37         22        41940458   \n",
       "MT-ND5                 4540     NaN         37          M           12384   \n",
       "MT-CO3                 4514     NaN         37          M            9531   \n",
       "\n",
       "             End_Position Strand              Consequence  \\\n",
       "Hugo_Symbol                                                 \n",
       "DVL1              1277461      +       synonymous_variant   \n",
       "AL590822.1        2144416      +         missense_variant   \n",
       "PLCH2             2435359      +      3_prime_UTR_variant   \n",
       "UBE4B            10177641      +         missense_variant   \n",
       "SRM              11119363      +       synonymous_variant   \n",
       "...                   ...    ...                      ...   \n",
       "MLLT4           168366689      +  downstream_gene_variant   \n",
       "MT-ND2               4605      +                      NaN   \n",
       "POLR3H           41940459      +    upstream_gene_variant   \n",
       "MT-ND5              12385      +                      NaN   \n",
       "MT-CO3               9532      +                      NaN   \n",
       "\n",
       "            Variant_Classification Variant_Type  ... RNAseq_AC  \\\n",
       "Hugo_Symbol                                      ...             \n",
       "DVL1                        Silent          SNP  ...       NaN   \n",
       "AL590822.1       Missense_Mutation          SNP  ...       NaN   \n",
       "PLCH2                        3'UTR          SNP  ...       NaN   \n",
       "UBE4B            Missense_Mutation          SNP  ...       NaN   \n",
       "SRM                         Silent          SNP  ...       NaN   \n",
       "...                            ...          ...  ...       ...   \n",
       "MLLT4                      3'Flank          DEL  ...     12:91   \n",
       "MT-ND2             Frame_Shift_Ins          INS  ...      9:18   \n",
       "POLR3H                     5'Flank          INS  ...       5:3   \n",
       "MT-ND5             Frame_Shift_Ins          INS  ...     12:69   \n",
       "MT-CO3             Frame_Shift_Ins          INS  ...     10:29   \n",
       "\n",
       "             SangerRecalibWES_AC SangerWES_AC TCGAhsCnt  WES_AC WGS_AC  \\\n",
       "Hugo_Symbol                                                              \n",
       "DVL1                         NaN          NaN         0   87:39    NaN   \n",
       "AL590822.1                   NaN          NaN         0   23:43    NaN   \n",
       "PLCH2                        NaN          NaN         0    9:25    NaN   \n",
       "UBE4B                        NaN          NaN         0   13:95    NaN   \n",
       "SRM                          NaN          NaN         0  113:57    NaN   \n",
       "...                          ...          ...       ...     ...    ...   \n",
       "MLLT4                        NaN          NaN         0     NaN    NaN   \n",
       "MT-ND2                       NaN          NaN         0     NaN    NaN   \n",
       "POLR3H                       NaN          NaN         0     NaN    NaN   \n",
       "MT-ND5                       NaN          NaN         0     NaN    NaN   \n",
       "MT-CO3                       NaN          NaN         0     NaN    NaN   \n",
       "\n",
       "                   cDNA_Change  isCOSMIChotspot  isDeleterious  isTCGAhotspot  \n",
       "Hugo_Symbol                                                                    \n",
       "DVL1                  c.438G>A            False          False          False  \n",
       "AL590822.1            c.604C>T            False          False          False  \n",
       "PLCH2                      NaN            False           True          False  \n",
       "UBE4B                 c.934G>A            False          False          False  \n",
       "SRM                   c.207C>A            False          False          False  \n",
       "...                        ...              ...            ...            ...  \n",
       "MLLT4        c.5198_5200delACA            False          False          False  \n",
       "MT-ND2           c.135_136insA            False           True          False  \n",
       "POLR3H                     NaN            False           True          False  \n",
       "MT-ND5             c.48_49insC            False           True          False  \n",
       "MT-CO3           c.325_326insC            False           True          False  \n",
       "\n",
       "[1212258 rows x 61 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "06a73912",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Column names:\n",
      "['Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome', 'Start_Position', 'End_Position', 'Strand', 'Consequence', 'Variant_Classification', 'Variant_Type', 'Reference_Allele', 'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'dbSNP_RS', 'dbSNP_Val_Status', 'Tumor_Sample_Barcode', 'Matched_Norm_Sample_Barcode', 'Match_Norm_Seq_Allele1', 'Match_Norm_Seq_Allele2', 'Tumor_Validation_Allele1', 'Tumor_Validation_Allele2', 'Match_Norm_Validation_Allele1', 'Match_Norm_Validation_Allele2', 'Verification_Status', 'Validation_Status', 'Mutation_Status', 'Sequencing_Phase', 'Sequence_Source', 'Validation_Method', 'Score', 'BAM_File', 'Sequencer', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count', 'HGVSc', 'HGVSp', 'HGVSp_Short', 'Transcript_ID', 'RefSeq', 'Protein_position', 'Codons', 'Hotspot', 'Broad_ID', 'COSMIChsCnt', 'Codon_Change', 'ExAC_AF', 'HC_AC', 'Protein_Change', 'RD_AC', 'RNAseq_AC', 'SangerRecalibWES_AC', 'SangerWES_AC', 'TCGAhsCnt', 'WES_AC', 'WGS_AC', 'cDNA_Change', 'isCOSMIChotspot', 'isDeleterious', 'isTCGAhotspot']\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>RNAseq_AC</th>\n",
       "      <th>SangerRecalibWES_AC</th>\n",
       "      <th>SangerWES_AC</th>\n",
       "      <th>TCGAhsCnt</th>\n",
       "      <th>WES_AC</th>\n",
       "      <th>WGS_AC</th>\n",
       "      <th>cDNA_Change</th>\n",
       "      <th>isCOSMIChotspot</th>\n",
       "      <th>isDeleterious</th>\n",
       "      <th>isTCGAhotspot</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7576854</td>\n",
       "      <td>7576854</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>88:33</td>\n",
       "      <td>23:50</td>\n",
       "      <td>24:52</td>\n",
       "      <td>0</td>\n",
       "      <td>30:33</td>\n",
       "      <td>18:15</td>\n",
       "      <td>c.992A&gt;G</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577120</td>\n",
       "      <td>7577120</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>30:1</td>\n",
       "      <td>30:1</td>\n",
       "      <td>181</td>\n",
       "      <td>205:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.818G&gt;A</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577536</td>\n",
       "      <td>7577536</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>249:1</td>\n",
       "      <td>23:0</td>\n",
       "      <td>17:0</td>\n",
       "      <td>19</td>\n",
       "      <td>337:0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.745A&gt;G</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578492</td>\n",
       "      <td>7578492</td>\n",
       "      <td>+</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>35:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>57:2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.438G&gt;A</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578400</td>\n",
       "      <td>7578400</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23:19</td>\n",
       "      <td>23:21</td>\n",
       "      <td>7</td>\n",
       "      <td>235:260</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.530C&gt;T</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578253</td>\n",
       "      <td>7578259</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>22:6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.590_596delTGGAAGG</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578259</td>\n",
       "      <td>7578260</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>INS</td>\n",
       "      <td>...</td>\n",
       "      <td>22:5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.589_590insCC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577150</td>\n",
       "      <td>7577159</td>\n",
       "      <td>+</td>\n",
       "      <td>splice_acceptor_variant,coding_sequence_varian...</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.783_788delGTAGTGGTAA</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578479</td>\n",
       "      <td>7578480</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.450_451delAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578483</td>\n",
       "      <td>7578484</td>\n",
       "      <td>+</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.446_447delCC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1218 rows × 61 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Entrez_Gene_Id  Center NCBI_Build Chromosome  Start_Position  \\\n",
       "Hugo_Symbol                                                                 \n",
       "TP53                   7157     NaN     GRCh37         17         7576854   \n",
       "TP53                   7157     NaN     GRCh37         17         7577120   \n",
       "TP53                   7157     NaN     GRCh37         17         7577536   \n",
       "TP53                   7157     NaN     GRCh37         17         7578492   \n",
       "TP53                   7157     NaN     GRCh37         17         7578400   \n",
       "...                     ...     ...        ...        ...             ...   \n",
       "TP53                   7157     NaN     GRCh37         17         7578253   \n",
       "TP53                   7157     NaN     GRCh37         17         7578259   \n",
       "TP53                   7157     NaN     GRCh37         17         7577150   \n",
       "TP53                   7157     NaN     GRCh37         17         7578479   \n",
       "TP53                   7157     NaN     GRCh37         17         7578483   \n",
       "\n",
       "             End_Position Strand  \\\n",
       "Hugo_Symbol                        \n",
       "TP53              7576854      +   \n",
       "TP53              7577120      +   \n",
       "TP53              7577536      +   \n",
       "TP53              7578492      +   \n",
       "TP53              7578400      +   \n",
       "...                   ...    ...   \n",
       "TP53              7578259      +   \n",
       "TP53              7578260      +   \n",
       "TP53              7577159      +   \n",
       "TP53              7578480      +   \n",
       "TP53              7578484      +   \n",
       "\n",
       "                                                   Consequence  \\\n",
       "Hugo_Symbol                                                      \n",
       "TP53                    missense_variant,splice_region_variant   \n",
       "TP53                                          missense_variant   \n",
       "TP53                                          missense_variant   \n",
       "TP53                                               stop_gained   \n",
       "TP53                                          missense_variant   \n",
       "...                                                        ...   \n",
       "TP53                                        frameshift_variant   \n",
       "TP53                                        frameshift_variant   \n",
       "TP53         splice_acceptor_variant,coding_sequence_varian...   \n",
       "TP53                                        frameshift_variant   \n",
       "TP53                                        frameshift_variant   \n",
       "\n",
       "            Variant_Classification Variant_Type  ... RNAseq_AC  \\\n",
       "Hugo_Symbol                                      ...             \n",
       "TP53             Missense_Mutation          SNP  ...     88:33   \n",
       "TP53             Missense_Mutation          SNP  ...       NaN   \n",
       "TP53             Missense_Mutation          SNP  ...     249:1   \n",
       "TP53             Nonsense_Mutation          SNP  ...      35:2   \n",
       "TP53             Missense_Mutation          SNP  ...       NaN   \n",
       "...                            ...          ...  ...       ...   \n",
       "TP53               Frame_Shift_Del          DEL  ...      22:6   \n",
       "TP53               Frame_Shift_Ins          INS  ...      22:5   \n",
       "TP53                   Splice_Site          DEL  ...       NaN   \n",
       "TP53               Frame_Shift_Del          DEL  ...       NaN   \n",
       "TP53               Frame_Shift_Del          DEL  ...       NaN   \n",
       "\n",
       "             SangerRecalibWES_AC SangerWES_AC TCGAhsCnt   WES_AC WGS_AC  \\\n",
       "Hugo_Symbol                                                               \n",
       "TP53                       23:50        24:52         0    30:33  18:15   \n",
       "TP53                        30:1         30:1       181    205:0    NaN   \n",
       "TP53                        23:0         17:0        19    337:0    NaN   \n",
       "TP53                         NaN          NaN         9     57:2    NaN   \n",
       "TP53                       23:19        23:21         7  235:260    NaN   \n",
       "...                          ...          ...       ...      ...    ...   \n",
       "TP53                         NaN          NaN         9      NaN    NaN   \n",
       "TP53                         NaN          NaN         5      NaN    NaN   \n",
       "TP53                         NaN          NaN         0      NaN    NaN   \n",
       "TP53                         NaN          NaN        22      NaN    NaN   \n",
       "TP53                         NaN          NaN         0      NaN    NaN   \n",
       "\n",
       "                        cDNA_Change  isCOSMIChotspot  isDeleterious  \\\n",
       "Hugo_Symbol                                                           \n",
       "TP53                       c.992A>G             True           True   \n",
       "TP53                       c.818G>A             True          False   \n",
       "TP53                       c.745A>G             True          False   \n",
       "TP53                       c.438G>A             True           True   \n",
       "TP53                       c.530C>T             True          False   \n",
       "...                             ...              ...            ...   \n",
       "TP53            c.590_596delTGGAAGG             True           True   \n",
       "TP53                 c.589_590insCC             True           True   \n",
       "TP53         c.783_788delGTAGTGGTAA             True           True   \n",
       "TP53                 c.450_451delAC             True           True   \n",
       "TP53                 c.446_447delCC             True           True   \n",
       "\n",
       "             isTCGAhotspot  \n",
       "Hugo_Symbol                 \n",
       "TP53                 False  \n",
       "TP53                  True  \n",
       "TP53                  True  \n",
       "TP53                  True  \n",
       "TP53                  True  \n",
       "...                    ...  \n",
       "TP53                  True  \n",
       "TP53                  True  \n",
       "TP53                 False  \n",
       "TP53                  True  \n",
       "TP53                 False  \n",
       "\n",
       "[1218 rows x 61 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Display the column names to understand the structure\n",
    "print(\"Column names:\")\n",
    "print(mutations.columns.tolist())\n",
    "\n",
    "# Filter for TP53 mutations\n",
    "# The gene symbol appears to be in the 'Hugo_Symbol' column based on the data\n",
    "tp53_mutations = mutations.loc['TP53']\n",
    "tp53_mutations\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "db1e0e2e",
   "metadata": {},
   "outputs": [],
   "source": [
    "tp_53 = tp53_mutations[['Tumor_Sample_Barcode', 'Variant_Type', 'Variant_Classification']].copy().reset_index().drop([\"Hugo_Symbol\"], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "06375835",
   "metadata": {},
   "outputs": [],
   "source": [
    "# First, let's prepare the tp_53 data by selecting unique sample information\n",
    "# Since there might be multiple mutations per sample, we need to handle duplicates\n",
    "tp53 = tp_53[['Tumor_Sample_Barcode', 'Variant_Type', 'Variant_Classification']].drop_duplicates().set_index(\"Tumor_Sample_Barcode\")\n",
    "tp53.index = tp53.index.str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e332dd0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "expr.index = expr.index.str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f913555c",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_data = expr.join(tp53)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "4920a724",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data shape:(1257, 56320)\n"
     ]
    }
   ],
   "source": [
    "print(f\"Data shape:{merged_data.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "b50fb662",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_data.Variant_Classification = merged_data.Variant_Classification.fillna(\"No_Mutation\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "b6b3e989",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Variant_Classification\n",
       "Missense_Mutation    540\n",
       "No_Mutation          402\n",
       "Nonsense_Mutation    103\n",
       "Splice_Site           74\n",
       "Frame_Shift_Del       71\n",
       "Frame_Shift_Ins       20\n",
       "Silent                17\n",
       "In_Frame_Del          15\n",
       "Splice_Region         12\n",
       "In_Frame_Ins           3\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_data.Variant_Classification.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "132ce3fe",
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_data.is_mutated = merged_data"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
