{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ed3c9ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import necessary libraries for data manipulation and encoding\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.preprocessing import LabelEncoder"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1dd3ddd1",
   "metadata": {},
   "source": [
    "**Dataset link:**\n",
    "\n",
    "https://www.cbioportal.org/results/oncoprint?cancer_study_list=pancan_pcawg_2020&Z_SCORE_THRESHOLD=2.0&RPPA_SCORE_THRESHOLD=2.0&profileFilter=mutations&case_set_id=pancan_pcawg_2020_sequenced&gene_list=TP53&geneset_list=%20&tab_index=tab_visualize&Action=Submit\n",
    "\n",
    "This dataset contains TCGA mRNA expression and mutation data for TP53."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "622de76c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define data directory and file paths\n",
    "DATA_DIR = \"../data/raw/tcga\"\n",
    "EXPR_FILE = f\"{DATA_DIR}/data_mrna_seq_fpkm.txt\"\n",
    "\n",
    "MUTATIONS_FILE=f\"{DATA_DIR}/data_mutations.txt\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dbe709a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load mRNA expression data\n",
    "expr = pd.read_csv(EXPR_FILE, sep='\\t', index_col=0)\n",
    "\n",
    "# Load mutations file (skip comment lines, set index)\n",
    "mutations = pd.read_csv(MUTATIONS_FILE,  sep='\\t', comment = '#' ,low_memory=False, index_col=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12f886af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>SP89389</th>\n",
       "      <th>SP21193</th>\n",
       "      <th>SP13206</th>\n",
       "      <th>SP103623</th>\n",
       "      <th>SP32742</th>\n",
       "      <th>SP111095</th>\n",
       "      <th>SP8394</th>\n",
       "      <th>SP87446</th>\n",
       "      <th>SP36586</th>\n",
       "      <th>SP123902</th>\n",
       "      <th>...</th>\n",
       "      <th>SP15656</th>\n",
       "      <th>SP123888</th>\n",
       "      <th>SP59420</th>\n",
       "      <th>SP116679</th>\n",
       "      <th>SP1377</th>\n",
       "      <th>SP16269</th>\n",
       "      <th>SP122676</th>\n",
       "      <th>SP88776</th>\n",
       "      <th>SP64546</th>\n",
       "      <th>SP21057</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TSPAN6</th>\n",
       "      <td>65.515874</td>\n",
       "      <td>93.507754</td>\n",
       "      <td>15.155525</td>\n",
       "      <td>9.751670</td>\n",
       "      <td>16.048850</td>\n",
       "      <td>22.724550</td>\n",
       "      <td>29.154328</td>\n",
       "      <td>18.058406</td>\n",
       "      <td>15.486367</td>\n",
       "      <td>43.248311</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>27.329458</td>\n",
       "      <td>3.633029</td>\n",
       "      <td>2.302684</td>\n",
       "      <td>227.790781</td>\n",
       "      <td>0.007476</td>\n",
       "      <td>9.736614</td>\n",
       "      <td>29.283034</td>\n",
       "      <td>28.045150</td>\n",
       "      <td>62.010994</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNMD</th>\n",
       "      <td>0.046881</td>\n",
       "      <td>0.655002</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.188735</td>\n",
       "      <td>0.014542</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.110858</td>\n",
       "      <td>0.031887</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.118371</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.033938</td>\n",
       "      <td>0.064275</td>\n",
       "      <td>0.414507</td>\n",
       "      <td>0.345589</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.062405</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.026696</td>\n",
       "      <td>0.405261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DPM1</th>\n",
       "      <td>63.315522</td>\n",
       "      <td>23.608843</td>\n",
       "      <td>42.136192</td>\n",
       "      <td>14.507132</td>\n",
       "      <td>45.302557</td>\n",
       "      <td>24.728471</td>\n",
       "      <td>22.578201</td>\n",
       "      <td>21.447736</td>\n",
       "      <td>19.492883</td>\n",
       "      <td>20.331784</td>\n",
       "      <td>...</td>\n",
       "      <td>18.197652</td>\n",
       "      <td>38.290606</td>\n",
       "      <td>35.459968</td>\n",
       "      <td>31.888606</td>\n",
       "      <td>34.589654</td>\n",
       "      <td>23.290497</td>\n",
       "      <td>31.444337</td>\n",
       "      <td>23.118048</td>\n",
       "      <td>51.549932</td>\n",
       "      <td>30.790944</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SCYL3</th>\n",
       "      <td>2.222857</td>\n",
       "      <td>1.933762</td>\n",
       "      <td>2.318698</td>\n",
       "      <td>2.342167</td>\n",
       "      <td>4.373694</td>\n",
       "      <td>4.050503</td>\n",
       "      <td>3.372809</td>\n",
       "      <td>2.359359</td>\n",
       "      <td>1.930601</td>\n",
       "      <td>1.886838</td>\n",
       "      <td>...</td>\n",
       "      <td>4.735389</td>\n",
       "      <td>1.356212</td>\n",
       "      <td>4.340347</td>\n",
       "      <td>5.257385</td>\n",
       "      <td>2.496924</td>\n",
       "      <td>6.466530</td>\n",
       "      <td>1.072155</td>\n",
       "      <td>3.699246</td>\n",
       "      <td>1.989817</td>\n",
       "      <td>3.313589</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>C1orf112</th>\n",
       "      <td>2.565836</td>\n",
       "      <td>1.861716</td>\n",
       "      <td>2.870440</td>\n",
       "      <td>0.454312</td>\n",
       "      <td>4.125029</td>\n",
       "      <td>1.877498</td>\n",
       "      <td>1.999630</td>\n",
       "      <td>0.747373</td>\n",
       "      <td>0.501911</td>\n",
       "      <td>0.369148</td>\n",
       "      <td>...</td>\n",
       "      <td>6.592257</td>\n",
       "      <td>0.259415</td>\n",
       "      <td>1.864762</td>\n",
       "      <td>2.262558</td>\n",
       "      <td>1.257206</td>\n",
       "      <td>6.384945</td>\n",
       "      <td>1.207673</td>\n",
       "      <td>0.539174</td>\n",
       "      <td>1.556905</td>\n",
       "      <td>1.757992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BX649553.1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BX649553.3</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BX649553.4</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RN7SL355P</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MIR3690</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>55921 rows × 1210 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               SP89389    SP21193    SP13206   SP103623    SP32742   SP111095  \\\n",
       "Hugo_Symbol                                                                     \n",
       "TSPAN6       65.515874  93.507754  15.155525   9.751670  16.048850  22.724550   \n",
       "TNMD          0.046881   0.655002   0.000000   0.188735   0.014542   0.000000   \n",
       "DPM1         63.315522  23.608843  42.136192  14.507132  45.302557  24.728471   \n",
       "SCYL3         2.222857   1.933762   2.318698   2.342167   4.373694   4.050503   \n",
       "C1orf112      2.565836   1.861716   2.870440   0.454312   4.125029   1.877498   \n",
       "...                ...        ...        ...        ...        ...        ...   \n",
       "BX649553.1    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "BX649553.3    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "BX649553.4    0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "RN7SL355P     0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "MIR3690       0.000000   0.000000   0.000000   0.000000   0.000000   0.000000   \n",
       "\n",
       "                SP8394    SP87446    SP36586   SP123902  ...    SP15656  \\\n",
       "Hugo_Symbol                                              ...              \n",
       "TSPAN6       29.154328  18.058406  15.486367  43.248311  ...   0.000000   \n",
       "TNMD          0.110858   0.031887   0.000000   0.118371  ...   0.000000   \n",
       "DPM1         22.578201  21.447736  19.492883  20.331784  ...  18.197652   \n",
       "SCYL3         3.372809   2.359359   1.930601   1.886838  ...   4.735389   \n",
       "C1orf112      1.999630   0.747373   0.501911   0.369148  ...   6.592257   \n",
       "...                ...        ...        ...        ...  ...        ...   \n",
       "BX649553.1    0.000000   0.000000   0.000000   0.000000  ...   0.000000   \n",
       "BX649553.3    0.000000   0.000000   0.000000   0.000000  ...   0.000000   \n",
       "BX649553.4    0.000000   0.000000   0.000000   0.000000  ...   0.000000   \n",
       "RN7SL355P     0.000000   0.000000   0.000000   0.000000  ...   0.000000   \n",
       "MIR3690       0.000000   0.000000   0.000000   0.000000  ...   0.000000   \n",
       "\n",
       "              SP123888    SP59420   SP116679      SP1377    SP16269  \\\n",
       "Hugo_Symbol                                                           \n",
       "TSPAN6       27.329458   3.633029   2.302684  227.790781   0.007476   \n",
       "TNMD          0.033938   0.064275   0.414507    0.345589   0.000000   \n",
       "DPM1         38.290606  35.459968  31.888606   34.589654  23.290497   \n",
       "SCYL3         1.356212   4.340347   5.257385    2.496924   6.466530   \n",
       "C1orf112      0.259415   1.864762   2.262558    1.257206   6.384945   \n",
       "...                ...        ...        ...         ...        ...   \n",
       "BX649553.1    0.000000   0.000000   0.000000    0.000000   0.000000   \n",
       "BX649553.3    0.000000   0.000000   0.000000    0.000000   0.000000   \n",
       "BX649553.4    0.000000   0.000000   0.000000    0.000000   0.000000   \n",
       "RN7SL355P     0.000000   0.000000   0.000000    0.000000   0.000000   \n",
       "MIR3690       0.000000   0.000000   0.000000    0.000000   0.000000   \n",
       "\n",
       "              SP122676    SP88776    SP64546    SP21057  \n",
       "Hugo_Symbol                                              \n",
       "TSPAN6        9.736614  29.283034  28.045150  62.010994  \n",
       "TNMD          0.062405   0.000000   0.026696   0.405261  \n",
       "DPM1         31.444337  23.118048  51.549932  30.790944  \n",
       "SCYL3         1.072155   3.699246   1.989817   3.313589  \n",
       "C1orf112      1.207673   0.539174   1.556905   1.757992  \n",
       "...                ...        ...        ...        ...  \n",
       "BX649553.1    0.000000   0.000000   0.000000   0.000000  \n",
       "BX649553.3    0.000000   0.000000   0.000000   0.000000  \n",
       "BX649553.4    0.000000   0.000000   0.000000   0.000000  \n",
       "RN7SL355P     0.000000   0.000000   0.000000   0.000000  \n",
       "MIR3690       0.000000   0.000000   0.000000   0.000000  \n",
       "\n",
       "[55921 rows x 1210 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview the first few rows of the expression data\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a145259d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>Annotation_Status</th>\n",
       "      <th>DNA_VAF</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>MATN1</th>\n",
       "      <td>4146.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>31189041</td>\n",
       "      <td>31189041</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000373765.4:c.922T&gt;C</td>\n",
       "      <td>p.Ser308Pro</td>\n",
       "      <td>p.S308P</td>\n",
       "      <td>ENST00000373765</td>\n",
       "      <td>NM_002379.3</td>\n",
       "      <td>308.0</td>\n",
       "      <td>Tca/Cca</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.1190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TFAP2E</th>\n",
       "      <td>339488.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>36055608</td>\n",
       "      <td>36055608</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000373235.3:c.863G&gt;A</td>\n",
       "      <td>p.Arg288His</td>\n",
       "      <td>p.R288H</td>\n",
       "      <td>ENST00000373235</td>\n",
       "      <td>NM_178548.3</td>\n",
       "      <td>288.0</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.0851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FGGY</th>\n",
       "      <td>55277.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>60228203</td>\n",
       "      <td>60228203</td>\n",
       "      <td>+</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>Silent</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000303721.7:c.1603C&gt;T</td>\n",
       "      <td>p.Leu535=</td>\n",
       "      <td>p.L535=</td>\n",
       "      <td>ENST00000303721</td>\n",
       "      <td>NM_018291.3</td>\n",
       "      <td>535.0</td>\n",
       "      <td>Ctg/Ttg</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.4103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PGM1</th>\n",
       "      <td>5236.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>64125315</td>\n",
       "      <td>64125315</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000371084.3:c.1658C&gt;T</td>\n",
       "      <td>p.Thr553Met</td>\n",
       "      <td>p.T553M</td>\n",
       "      <td>ENST00000371084</td>\n",
       "      <td>NM_002633.2</td>\n",
       "      <td>553.0</td>\n",
       "      <td>aCg/aTg</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.1400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DEPDC1</th>\n",
       "      <td>55635.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>1</td>\n",
       "      <td>68948436</td>\n",
       "      <td>68948436</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000456315.2:c.1055G&gt;T</td>\n",
       "      <td>p.Arg352Ile</td>\n",
       "      <td>p.R352I</td>\n",
       "      <td>ENST00000456315</td>\n",
       "      <td>NM_001114120.1</td>\n",
       "      <td>352.0</td>\n",
       "      <td>aGa/aTa</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.1765</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Entrez_Gene_Id  Center NCBI_Build Chromosome  Start_Position  \\\n",
       "Hugo_Symbol                                                                 \n",
       "MATN1                4146.0     NaN     GRCh37          1        31189041   \n",
       "TFAP2E             339488.0     NaN     GRCh37          1        36055608   \n",
       "FGGY                55277.0     NaN     GRCh37          1        60228203   \n",
       "PGM1                 5236.0     NaN     GRCh37          1        64125315   \n",
       "DEPDC1              55635.0     NaN     GRCh37          1        68948436   \n",
       "\n",
       "             End_Position Strand         Consequence Variant_Classification  \\\n",
       "Hugo_Symbol                                                                   \n",
       "MATN1            31189041      +    missense_variant      Missense_Mutation   \n",
       "TFAP2E           36055608      +    missense_variant      Missense_Mutation   \n",
       "FGGY             60228203      +  synonymous_variant                 Silent   \n",
       "PGM1             64125315      +    missense_variant      Missense_Mutation   \n",
       "DEPDC1           68948436      +    missense_variant      Missense_Mutation   \n",
       "\n",
       "            Variant_Type  ...                        HGVSc        HGVSp  \\\n",
       "Hugo_Symbol               ...                                             \n",
       "MATN1                SNP  ...   ENST00000373765.4:c.922T>C  p.Ser308Pro   \n",
       "TFAP2E               SNP  ...   ENST00000373235.3:c.863G>A  p.Arg288His   \n",
       "FGGY                 SNP  ...  ENST00000303721.7:c.1603C>T    p.Leu535=   \n",
       "PGM1                 SNP  ...  ENST00000371084.3:c.1658C>T  p.Thr553Met   \n",
       "DEPDC1               SNP  ...  ENST00000456315.2:c.1055G>T  p.Arg352Ile   \n",
       "\n",
       "            HGVSp_Short    Transcript_ID          RefSeq Protein_position  \\\n",
       "Hugo_Symbol                                                                 \n",
       "MATN1           p.S308P  ENST00000373765     NM_002379.3            308.0   \n",
       "TFAP2E          p.R288H  ENST00000373235     NM_178548.3            288.0   \n",
       "FGGY            p.L535=  ENST00000303721     NM_018291.3            535.0   \n",
       "PGM1            p.T553M  ENST00000371084     NM_002633.2            553.0   \n",
       "DEPDC1          p.R352I  ENST00000456315  NM_001114120.1            352.0   \n",
       "\n",
       "              Codons  Hotspot  Annotation_Status  DNA_VAF  \n",
       "Hugo_Symbol                                                \n",
       "MATN1        Tca/Cca        0            SUCCESS   0.1190  \n",
       "TFAP2E       cGc/cAc        0            SUCCESS   0.0851  \n",
       "FGGY         Ctg/Ttg        0            SUCCESS   0.4103  \n",
       "PGM1         aCg/aTg        0            SUCCESS   0.1400  \n",
       "DEPDC1       aGa/aTa        0            SUCCESS   0.1765  \n",
       "\n",
       "[5 rows x 46 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview the first few rows of the mutations data\n",
    "mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c03a013",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>...</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Hotspot</th>\n",
       "      <th>Annotation_Status</th>\n",
       "      <th>DNA_VAF</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577058</td>\n",
       "      <td>7577058</td>\n",
       "      <td>+</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.880G&gt;T</td>\n",
       "      <td>p.Glu294Ter</td>\n",
       "      <td>p.E294*</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>294.0</td>\n",
       "      <td>Gag/Tag</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.8846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578475</td>\n",
       "      <td>7578475</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.455C&gt;T</td>\n",
       "      <td>p.Pro152Leu</td>\n",
       "      <td>p.P152L</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>152.0</td>\n",
       "      <td>cCg/cTg</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.1154</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577560</td>\n",
       "      <td>7577560</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.721T&gt;C</td>\n",
       "      <td>p.Ser241Pro</td>\n",
       "      <td>p.S241P</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>241.0</td>\n",
       "      <td>Tcc/Ccc</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.9375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578540</td>\n",
       "      <td>7578548</td>\n",
       "      <td>+</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>In_Frame_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.382_390del</td>\n",
       "      <td>p.Pro128_Leu130del</td>\n",
       "      <td>p.P128_L130del</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>128.0</td>\n",
       "      <td>CCTGCCCTC/-</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.8461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577580</td>\n",
       "      <td>7577580</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.701A&gt;G</td>\n",
       "      <td>p.Tyr234Cys</td>\n",
       "      <td>p.Y234C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>234.0</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.4524</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577548</td>\n",
       "      <td>7577548</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.733G&gt;A</td>\n",
       "      <td>p.Gly245Ser</td>\n",
       "      <td>p.G245S</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>245.0</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.5000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578555</td>\n",
       "      <td>7578555</td>\n",
       "      <td>+</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.376-1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.X126_splice</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>126.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>1.0000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577557</td>\n",
       "      <td>7577557</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.724T&gt;A</td>\n",
       "      <td>p.Cys242Ser</td>\n",
       "      <td>p.C242S</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>242.0</td>\n",
       "      <td>Tgc/Agc</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.3214</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578212</td>\n",
       "      <td>7578212</td>\n",
       "      <td>+</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.637C&gt;T</td>\n",
       "      <td>p.Arg213Ter</td>\n",
       "      <td>p.R213*</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>213.0</td>\n",
       "      <td>Cga/Tga</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.3913</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TP53</th>\n",
       "      <td>7157.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578235</td>\n",
       "      <td>7578235</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>...</td>\n",
       "      <td>ENST00000269305.4:c.614A&gt;G</td>\n",
       "      <td>p.Tyr205Cys</td>\n",
       "      <td>p.Y205C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>NM_001126112.2</td>\n",
       "      <td>205.0</td>\n",
       "      <td>tAt/tGt</td>\n",
       "      <td>0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>0.4342</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>941 rows × 46 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Entrez_Gene_Id  Center NCBI_Build Chromosome  Start_Position  \\\n",
       "Hugo_Symbol                                                                 \n",
       "TP53                 7157.0     NaN     GRCh37         17         7577058   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7578475   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7577560   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7578540   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7577580   \n",
       "...                     ...     ...        ...        ...             ...   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7577548   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7578555   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7577557   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7578212   \n",
       "TP53                 7157.0     NaN     GRCh37         17         7578235   \n",
       "\n",
       "             End_Position Strand              Consequence  \\\n",
       "Hugo_Symbol                                                 \n",
       "TP53              7577058      +              stop_gained   \n",
       "TP53              7578475      +         missense_variant   \n",
       "TP53              7577560      +         missense_variant   \n",
       "TP53              7578548      +         inframe_deletion   \n",
       "TP53              7577580      +         missense_variant   \n",
       "...                   ...    ...                      ...   \n",
       "TP53              7577548      +         missense_variant   \n",
       "TP53              7578555      +  splice_acceptor_variant   \n",
       "TP53              7577557      +         missense_variant   \n",
       "TP53              7578212      +              stop_gained   \n",
       "TP53              7578235      +         missense_variant   \n",
       "\n",
       "            Variant_Classification Variant_Type  ...  \\\n",
       "Hugo_Symbol                                      ...   \n",
       "TP53             Nonsense_Mutation          SNP  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "TP53                  In_Frame_Del          DEL  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "...                            ...          ...  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "TP53                   Splice_Site          SNP  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "TP53             Nonsense_Mutation          SNP  ...   \n",
       "TP53             Missense_Mutation          SNP  ...   \n",
       "\n",
       "                                      HGVSc               HGVSp  \\\n",
       "Hugo_Symbol                                                       \n",
       "TP53             ENST00000269305.4:c.880G>T         p.Glu294Ter   \n",
       "TP53             ENST00000269305.4:c.455C>T         p.Pro152Leu   \n",
       "TP53             ENST00000269305.4:c.721T>C         p.Ser241Pro   \n",
       "TP53         ENST00000269305.4:c.382_390del  p.Pro128_Leu130del   \n",
       "TP53             ENST00000269305.4:c.701A>G         p.Tyr234Cys   \n",
       "...                                     ...                 ...   \n",
       "TP53             ENST00000269305.4:c.733G>A         p.Gly245Ser   \n",
       "TP53           ENST00000269305.4:c.376-1G>A                 NaN   \n",
       "TP53             ENST00000269305.4:c.724T>A         p.Cys242Ser   \n",
       "TP53             ENST00000269305.4:c.637C>T         p.Arg213Ter   \n",
       "TP53             ENST00000269305.4:c.614A>G         p.Tyr205Cys   \n",
       "\n",
       "                HGVSp_Short    Transcript_ID          RefSeq Protein_position  \\\n",
       "Hugo_Symbol                                                                     \n",
       "TP53                p.E294*  ENST00000269305  NM_001126112.2            294.0   \n",
       "TP53                p.P152L  ENST00000269305  NM_001126112.2            152.0   \n",
       "TP53                p.S241P  ENST00000269305  NM_001126112.2            241.0   \n",
       "TP53         p.P128_L130del  ENST00000269305  NM_001126112.2            128.0   \n",
       "TP53                p.Y234C  ENST00000269305  NM_001126112.2            234.0   \n",
       "...                     ...              ...             ...              ...   \n",
       "TP53                p.G245S  ENST00000269305  NM_001126112.2            245.0   \n",
       "TP53          p.X126_splice  ENST00000269305  NM_001126112.2            126.0   \n",
       "TP53                p.C242S  ENST00000269305  NM_001126112.2            242.0   \n",
       "TP53                p.R213*  ENST00000269305  NM_001126112.2            213.0   \n",
       "TP53                p.Y205C  ENST00000269305  NM_001126112.2            205.0   \n",
       "\n",
       "                  Codons  Hotspot  Annotation_Status  DNA_VAF  \n",
       "Hugo_Symbol                                                    \n",
       "TP53             Gag/Tag        0            SUCCESS   0.8846  \n",
       "TP53             cCg/cTg        0            SUCCESS   0.1154  \n",
       "TP53             Tcc/Ccc        0            SUCCESS   0.9375  \n",
       "TP53         CCTGCCCTC/-        0            SUCCESS   0.8461  \n",
       "TP53             tAc/tGc        0            SUCCESS   0.4524  \n",
       "...                  ...      ...                ...      ...  \n",
       "TP53             Ggc/Agc        0            SUCCESS   0.5000  \n",
       "TP53                 NaN        0            SUCCESS   1.0000  \n",
       "TP53             Tgc/Agc        0            SUCCESS   0.3214  \n",
       "TP53             Cga/Tga        0            SUCCESS   0.3913  \n",
       "TP53             tAt/tGt        0            SUCCESS   0.4342  \n",
       "\n",
       "[941 rows x 46 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Extract TP53 mutation data from the mutations dataframe\n",
    "# Select relevant columns and reset index\n",
    "# Drop the 'Hugo_Symbol' column as it's redundant after filtering\n",
    "\n",
    "tp53_mutations = mutations.loc['TP53']\n",
    "tp_53 = tp53_mutations[['Tumor_Sample_Barcode', 'Variant_Type', 'Variant_Classification']].copy().reset_index().drop([\"Hugo_Symbol\"], axis=1)\n",
    "# Preview the TP53 mutations data\n",
    "tp53_mutations.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb5a529a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SP101724</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SP22031</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SP59388</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SP94588</td>\n",
       "      <td>DEL</td>\n",
       "      <td>In_Frame_Del</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SP7692</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>936</th>\n",
       "      <td>SP111033</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>937</th>\n",
       "      <td>SP24815</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Splice_Site</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>938</th>\n",
       "      <td>SP112245</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>939</th>\n",
       "      <td>SP124415</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>940</th>\n",
       "      <td>SP124415</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>941 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Tumor_Sample_Barcode Variant_Type Variant_Classification\n",
       "0               SP101724          SNP      Nonsense_Mutation\n",
       "1                SP22031          SNP      Missense_Mutation\n",
       "2                SP59388          SNP      Missense_Mutation\n",
       "3                SP94588          DEL           In_Frame_Del\n",
       "4                 SP7692          SNP      Missense_Mutation\n",
       "..                   ...          ...                    ...\n",
       "936             SP111033          SNP      Missense_Mutation\n",
       "937              SP24815          SNP            Splice_Site\n",
       "938             SP112245          SNP      Missense_Mutation\n",
       "939             SP124415          SNP      Nonsense_Mutation\n",
       "940             SP124415          SNP      Missense_Mutation\n",
       "\n",
       "[941 rows x 3 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview the first few rows of the TP53 mutation subset\n",
    "tp_53.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a4fcb7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Prepare the tp_53 data by selecting unique sample information\n",
    "# Handle duplicates by keeping only unique sample-mutation pairs\n",
    "# Set Tumor_Sample_Barcode as the index and strip whitespace\n",
    "tp53 = tp_53[['Tumor_Sample_Barcode', 'Variant_Type', 'Variant_Classification']].drop_duplicates().set_index(\"Tumor_Sample_Barcode\")\n",
    "tp53.index = tp53.index.str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc75d949",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>TSPAN6</th>\n",
       "      <th>TNMD</th>\n",
       "      <th>DPM1</th>\n",
       "      <th>SCYL3</th>\n",
       "      <th>C1orf112</th>\n",
       "      <th>FGR</th>\n",
       "      <th>CFH</th>\n",
       "      <th>FUCA2</th>\n",
       "      <th>GCLC</th>\n",
       "      <th>NFYA</th>\n",
       "      <th>...</th>\n",
       "      <th>AP000230.1</th>\n",
       "      <th>RP11-80H18.4</th>\n",
       "      <th>RP13-297E16.4</th>\n",
       "      <th>LL0YNC03-29C1.1</th>\n",
       "      <th>RP13-297E16.5</th>\n",
       "      <th>BX649553.1</th>\n",
       "      <th>BX649553.3</th>\n",
       "      <th>BX649553.4</th>\n",
       "      <th>RN7SL355P</th>\n",
       "      <th>MIR3690</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SP89389</th>\n",
       "      <td>65.515874</td>\n",
       "      <td>0.046881</td>\n",
       "      <td>63.315522</td>\n",
       "      <td>2.222857</td>\n",
       "      <td>2.565836</td>\n",
       "      <td>2.167233</td>\n",
       "      <td>1.265078</td>\n",
       "      <td>34.199996</td>\n",
       "      <td>4.974365</td>\n",
       "      <td>29.891291</td>\n",
       "      <td>...</td>\n",
       "      <td>0.147707</td>\n",
       "      <td>0.245060</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP21193</th>\n",
       "      <td>93.507754</td>\n",
       "      <td>0.655002</td>\n",
       "      <td>23.608843</td>\n",
       "      <td>1.933762</td>\n",
       "      <td>1.861716</td>\n",
       "      <td>3.873639</td>\n",
       "      <td>3.268175</td>\n",
       "      <td>23.050032</td>\n",
       "      <td>4.019948</td>\n",
       "      <td>8.379595</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011216</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP13206</th>\n",
       "      <td>15.155525</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>42.136192</td>\n",
       "      <td>2.318698</td>\n",
       "      <td>2.870440</td>\n",
       "      <td>4.286835</td>\n",
       "      <td>9.066806</td>\n",
       "      <td>32.926751</td>\n",
       "      <td>3.909932</td>\n",
       "      <td>9.277749</td>\n",
       "      <td>...</td>\n",
       "      <td>0.027060</td>\n",
       "      <td>0.179581</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP103623</th>\n",
       "      <td>9.751670</td>\n",
       "      <td>0.188735</td>\n",
       "      <td>14.507132</td>\n",
       "      <td>2.342167</td>\n",
       "      <td>0.454312</td>\n",
       "      <td>13.098299</td>\n",
       "      <td>26.108567</td>\n",
       "      <td>25.074303</td>\n",
       "      <td>0.682193</td>\n",
       "      <td>8.342081</td>\n",
       "      <td>...</td>\n",
       "      <td>0.241574</td>\n",
       "      <td>1.479853</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP32742</th>\n",
       "      <td>16.048850</td>\n",
       "      <td>0.014542</td>\n",
       "      <td>45.302557</td>\n",
       "      <td>4.373694</td>\n",
       "      <td>4.125029</td>\n",
       "      <td>4.316576</td>\n",
       "      <td>16.914069</td>\n",
       "      <td>15.602162</td>\n",
       "      <td>8.076706</td>\n",
       "      <td>10.929168</td>\n",
       "      <td>...</td>\n",
       "      <td>0.040090</td>\n",
       "      <td>0.152030</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP16269</th>\n",
       "      <td>0.007476</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>23.290497</td>\n",
       "      <td>6.466530</td>\n",
       "      <td>6.384945</td>\n",
       "      <td>235.016264</td>\n",
       "      <td>0.013622</td>\n",
       "      <td>5.548667</td>\n",
       "      <td>13.125604</td>\n",
       "      <td>17.631857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.432225</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP122676</th>\n",
       "      <td>9.736614</td>\n",
       "      <td>0.062405</td>\n",
       "      <td>31.444337</td>\n",
       "      <td>1.072155</td>\n",
       "      <td>1.207673</td>\n",
       "      <td>5.166039</td>\n",
       "      <td>5.904768</td>\n",
       "      <td>61.047477</td>\n",
       "      <td>2.502007</td>\n",
       "      <td>12.565596</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.733968</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP88776</th>\n",
       "      <td>29.283034</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>23.118048</td>\n",
       "      <td>3.699246</td>\n",
       "      <td>0.539174</td>\n",
       "      <td>0.450032</td>\n",
       "      <td>0.403973</td>\n",
       "      <td>25.097563</td>\n",
       "      <td>6.154079</td>\n",
       "      <td>9.255289</td>\n",
       "      <td>...</td>\n",
       "      <td>0.751576</td>\n",
       "      <td>1.280037</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP64546</th>\n",
       "      <td>28.045150</td>\n",
       "      <td>0.026696</td>\n",
       "      <td>51.549932</td>\n",
       "      <td>1.989817</td>\n",
       "      <td>1.556905</td>\n",
       "      <td>1.276369</td>\n",
       "      <td>9.101945</td>\n",
       "      <td>15.649566</td>\n",
       "      <td>6.412556</td>\n",
       "      <td>6.883272</td>\n",
       "      <td>...</td>\n",
       "      <td>0.087614</td>\n",
       "      <td>0.232576</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP21057</th>\n",
       "      <td>62.010994</td>\n",
       "      <td>0.405261</td>\n",
       "      <td>30.790944</td>\n",
       "      <td>3.313589</td>\n",
       "      <td>1.757992</td>\n",
       "      <td>1.442420</td>\n",
       "      <td>1.695267</td>\n",
       "      <td>28.040086</td>\n",
       "      <td>6.889364</td>\n",
       "      <td>9.635526</td>\n",
       "      <td>...</td>\n",
       "      <td>0.076611</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1210 rows × 55921 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "Hugo_Symbol              TSPAN6      TNMD       DPM1     SCYL3  C1orf112  \\\n",
       "Tumor_Sample_Barcode                                                       \n",
       "SP89389               65.515874  0.046881  63.315522  2.222857  2.565836   \n",
       "SP21193               93.507754  0.655002  23.608843  1.933762  1.861716   \n",
       "SP13206               15.155525  0.000000  42.136192  2.318698  2.870440   \n",
       "SP103623               9.751670  0.188735  14.507132  2.342167  0.454312   \n",
       "SP32742               16.048850  0.014542  45.302557  4.373694  4.125029   \n",
       "...                         ...       ...        ...       ...       ...   \n",
       "SP16269                0.007476  0.000000  23.290497  6.466530  6.384945   \n",
       "SP122676               9.736614  0.062405  31.444337  1.072155  1.207673   \n",
       "SP88776               29.283034  0.000000  23.118048  3.699246  0.539174   \n",
       "SP64546               28.045150  0.026696  51.549932  1.989817  1.556905   \n",
       "SP21057               62.010994  0.405261  30.790944  3.313589  1.757992   \n",
       "\n",
       "Hugo_Symbol                  FGR        CFH      FUCA2       GCLC       NFYA  \\\n",
       "Tumor_Sample_Barcode                                                           \n",
       "SP89389                 2.167233   1.265078  34.199996   4.974365  29.891291   \n",
       "SP21193                 3.873639   3.268175  23.050032   4.019948   8.379595   \n",
       "SP13206                 4.286835   9.066806  32.926751   3.909932   9.277749   \n",
       "SP103623               13.098299  26.108567  25.074303   0.682193   8.342081   \n",
       "SP32742                 4.316576  16.914069  15.602162   8.076706  10.929168   \n",
       "...                          ...        ...        ...        ...        ...   \n",
       "SP16269               235.016264   0.013622   5.548667  13.125604  17.631857   \n",
       "SP122676                5.166039   5.904768  61.047477   2.502007  12.565596   \n",
       "SP88776                 0.450032   0.403973  25.097563   6.154079   9.255289   \n",
       "SP64546                 1.276369   9.101945  15.649566   6.412556   6.883272   \n",
       "SP21057                 1.442420   1.695267  28.040086   6.889364   9.635526   \n",
       "\n",
       "Hugo_Symbol           ...  AP000230.1  RP11-80H18.4  RP13-297E16.4  \\\n",
       "Tumor_Sample_Barcode  ...                                            \n",
       "SP89389               ...    0.147707      0.245060            0.0   \n",
       "SP21193               ...    0.011216      0.000000            0.0   \n",
       "SP13206               ...    0.027060      0.179581            0.0   \n",
       "SP103623              ...    0.241574      1.479853            0.0   \n",
       "SP32742               ...    0.040090      0.152030            0.0   \n",
       "...                   ...         ...           ...            ...   \n",
       "SP16269               ...    0.000000      0.432225            0.0   \n",
       "SP122676              ...    0.000000      0.733968            0.0   \n",
       "SP88776               ...    0.751576      1.280037            0.0   \n",
       "SP64546               ...    0.087614      0.232576            0.0   \n",
       "SP21057               ...    0.076611      0.000000            0.0   \n",
       "\n",
       "Hugo_Symbol           LL0YNC03-29C1.1  RP13-297E16.5  BX649553.1  BX649553.3  \\\n",
       "Tumor_Sample_Barcode                                                           \n",
       "SP89389                           0.0            0.0         0.0         0.0   \n",
       "SP21193                           0.0            0.0         0.0         0.0   \n",
       "SP13206                           0.0            0.0         0.0         0.0   \n",
       "SP103623                          0.0            0.0         0.0         0.0   \n",
       "SP32742                           0.0            0.0         0.0         0.0   \n",
       "...                               ...            ...         ...         ...   \n",
       "SP16269                           0.0            0.0         0.0         0.0   \n",
       "SP122676                          0.0            0.0         0.0         0.0   \n",
       "SP88776                           0.0            0.0         0.0         0.0   \n",
       "SP64546                           0.0            0.0         0.0         0.0   \n",
       "SP21057                           0.0            0.0         0.0         0.0   \n",
       "\n",
       "Hugo_Symbol           BX649553.4  RN7SL355P  MIR3690  \n",
       "Tumor_Sample_Barcode                                  \n",
       "SP89389                      0.0        0.0      0.0  \n",
       "SP21193                      0.0        0.0      0.0  \n",
       "SP13206                      0.0        0.0      0.0  \n",
       "SP103623                     0.0        0.0      0.0  \n",
       "SP32742                      0.0        0.0      0.0  \n",
       "...                          ...        ...      ...  \n",
       "SP16269                      0.0        0.0      0.0  \n",
       "SP122676                     0.0        0.0      0.0  \n",
       "SP88776                      0.0        0.0      0.0  \n",
       "SP64546                      0.0        0.0      0.0  \n",
       "SP21057                      0.0        0.0      0.0  \n",
       "\n",
       "[1210 rows x 55921 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Transpose the expression dataframe so samples are rows\n",
    "expr = expr.transpose()\n",
    "expr.index.name = 'Tumor_Sample_Barcode'\n",
    "# Preview the transposed expression data\n",
    "expr.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "05866e24",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Strip whitespace from sample barcodes in the expression data\n",
    "expr.index = expr.index.str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5ee994d6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Merge expression data with TP53 mutation data on Tumor_Sample_Barcode\n",
    "merged_data = expr.join(tp53)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0425d752",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSPAN6</th>\n",
       "      <th>TNMD</th>\n",
       "      <th>DPM1</th>\n",
       "      <th>SCYL3</th>\n",
       "      <th>C1orf112</th>\n",
       "      <th>FGR</th>\n",
       "      <th>CFH</th>\n",
       "      <th>FUCA2</th>\n",
       "      <th>GCLC</th>\n",
       "      <th>NFYA</th>\n",
       "      <th>...</th>\n",
       "      <th>RP13-297E16.4</th>\n",
       "      <th>LL0YNC03-29C1.1</th>\n",
       "      <th>RP13-297E16.5</th>\n",
       "      <th>BX649553.1</th>\n",
       "      <th>BX649553.3</th>\n",
       "      <th>BX649553.4</th>\n",
       "      <th>RN7SL355P</th>\n",
       "      <th>MIR3690</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Variant_Classification</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>SP89389</th>\n",
       "      <td>65.515874</td>\n",
       "      <td>0.046881</td>\n",
       "      <td>63.315522</td>\n",
       "      <td>2.222857</td>\n",
       "      <td>2.565836</td>\n",
       "      <td>2.167233</td>\n",
       "      <td>1.265078</td>\n",
       "      <td>34.199996</td>\n",
       "      <td>4.974365</td>\n",
       "      <td>29.891291</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP21193</th>\n",
       "      <td>93.507754</td>\n",
       "      <td>0.655002</td>\n",
       "      <td>23.608843</td>\n",
       "      <td>1.933762</td>\n",
       "      <td>1.861716</td>\n",
       "      <td>3.873639</td>\n",
       "      <td>3.268175</td>\n",
       "      <td>23.050032</td>\n",
       "      <td>4.019948</td>\n",
       "      <td>8.379595</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP13206</th>\n",
       "      <td>15.155525</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>42.136192</td>\n",
       "      <td>2.318698</td>\n",
       "      <td>2.870440</td>\n",
       "      <td>4.286835</td>\n",
       "      <td>9.066806</td>\n",
       "      <td>32.926751</td>\n",
       "      <td>3.909932</td>\n",
       "      <td>9.277749</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP103623</th>\n",
       "      <td>9.751670</td>\n",
       "      <td>0.188735</td>\n",
       "      <td>14.507132</td>\n",
       "      <td>2.342167</td>\n",
       "      <td>0.454312</td>\n",
       "      <td>13.098299</td>\n",
       "      <td>26.108567</td>\n",
       "      <td>25.074303</td>\n",
       "      <td>0.682193</td>\n",
       "      <td>8.342081</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP32742</th>\n",
       "      <td>16.048850</td>\n",
       "      <td>0.014542</td>\n",
       "      <td>45.302557</td>\n",
       "      <td>4.373694</td>\n",
       "      <td>4.125029</td>\n",
       "      <td>4.316576</td>\n",
       "      <td>16.914069</td>\n",
       "      <td>15.602162</td>\n",
       "      <td>8.076706</td>\n",
       "      <td>10.929168</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP16269</th>\n",
       "      <td>0.007476</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>23.290497</td>\n",
       "      <td>6.466530</td>\n",
       "      <td>6.384945</td>\n",
       "      <td>235.016264</td>\n",
       "      <td>0.013622</td>\n",
       "      <td>5.548667</td>\n",
       "      <td>13.125604</td>\n",
       "      <td>17.631857</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP122676</th>\n",
       "      <td>9.736614</td>\n",
       "      <td>0.062405</td>\n",
       "      <td>31.444337</td>\n",
       "      <td>1.072155</td>\n",
       "      <td>1.207673</td>\n",
       "      <td>5.166039</td>\n",
       "      <td>5.904768</td>\n",
       "      <td>61.047477</td>\n",
       "      <td>2.502007</td>\n",
       "      <td>12.565596</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP88776</th>\n",
       "      <td>29.283034</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>23.118048</td>\n",
       "      <td>3.699246</td>\n",
       "      <td>0.539174</td>\n",
       "      <td>0.450032</td>\n",
       "      <td>0.403973</td>\n",
       "      <td>25.097563</td>\n",
       "      <td>6.154079</td>\n",
       "      <td>9.255289</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP64546</th>\n",
       "      <td>28.045150</td>\n",
       "      <td>0.026696</td>\n",
       "      <td>51.549932</td>\n",
       "      <td>1.989817</td>\n",
       "      <td>1.556905</td>\n",
       "      <td>1.276369</td>\n",
       "      <td>9.101945</td>\n",
       "      <td>15.649566</td>\n",
       "      <td>6.412556</td>\n",
       "      <td>6.883272</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SP21057</th>\n",
       "      <td>62.010994</td>\n",
       "      <td>0.405261</td>\n",
       "      <td>30.790944</td>\n",
       "      <td>3.313589</td>\n",
       "      <td>1.757992</td>\n",
       "      <td>1.442420</td>\n",
       "      <td>1.695267</td>\n",
       "      <td>28.040086</td>\n",
       "      <td>6.889364</td>\n",
       "      <td>9.635526</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>DEL</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1222 rows × 55923 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         TSPAN6      TNMD       DPM1     SCYL3  C1orf112  \\\n",
       "Tumor_Sample_Barcode                                                       \n",
       "SP89389               65.515874  0.046881  63.315522  2.222857  2.565836   \n",
       "SP21193               93.507754  0.655002  23.608843  1.933762  1.861716   \n",
       "SP13206               15.155525  0.000000  42.136192  2.318698  2.870440   \n",
       "SP103623               9.751670  0.188735  14.507132  2.342167  0.454312   \n",
       "SP32742               16.048850  0.014542  45.302557  4.373694  4.125029   \n",
       "...                         ...       ...        ...       ...       ...   \n",
       "SP16269                0.007476  0.000000  23.290497  6.466530  6.384945   \n",
       "SP122676               9.736614  0.062405  31.444337  1.072155  1.207673   \n",
       "SP88776               29.283034  0.000000  23.118048  3.699246  0.539174   \n",
       "SP64546               28.045150  0.026696  51.549932  1.989817  1.556905   \n",
       "SP21057               62.010994  0.405261  30.790944  3.313589  1.757992   \n",
       "\n",
       "                             FGR        CFH      FUCA2       GCLC       NFYA  \\\n",
       "Tumor_Sample_Barcode                                                           \n",
       "SP89389                 2.167233   1.265078  34.199996   4.974365  29.891291   \n",
       "SP21193                 3.873639   3.268175  23.050032   4.019948   8.379595   \n",
       "SP13206                 4.286835   9.066806  32.926751   3.909932   9.277749   \n",
       "SP103623               13.098299  26.108567  25.074303   0.682193   8.342081   \n",
       "SP32742                 4.316576  16.914069  15.602162   8.076706  10.929168   \n",
       "...                          ...        ...        ...        ...        ...   \n",
       "SP16269               235.016264   0.013622   5.548667  13.125604  17.631857   \n",
       "SP122676                5.166039   5.904768  61.047477   2.502007  12.565596   \n",
       "SP88776                 0.450032   0.403973  25.097563   6.154079   9.255289   \n",
       "SP64546                 1.276369   9.101945  15.649566   6.412556   6.883272   \n",
       "SP21057                 1.442420   1.695267  28.040086   6.889364   9.635526   \n",
       "\n",
       "                      ...  RP13-297E16.4  LL0YNC03-29C1.1  RP13-297E16.5  \\\n",
       "Tumor_Sample_Barcode  ...                                                  \n",
       "SP89389               ...            0.0              0.0            0.0   \n",
       "SP21193               ...            0.0              0.0            0.0   \n",
       "SP13206               ...            0.0              0.0            0.0   \n",
       "SP103623              ...            0.0              0.0            0.0   \n",
       "SP32742               ...            0.0              0.0            0.0   \n",
       "...                   ...            ...              ...            ...   \n",
       "SP16269               ...            0.0              0.0            0.0   \n",
       "SP122676              ...            0.0              0.0            0.0   \n",
       "SP88776               ...            0.0              0.0            0.0   \n",
       "SP64546               ...            0.0              0.0            0.0   \n",
       "SP21057               ...            0.0              0.0            0.0   \n",
       "\n",
       "                      BX649553.1  BX649553.3  BX649553.4  RN7SL355P  MIR3690  \\\n",
       "Tumor_Sample_Barcode                                                           \n",
       "SP89389                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP21193                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP13206                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP103623                     0.0         0.0         0.0        0.0      0.0   \n",
       "SP32742                      0.0         0.0         0.0        0.0      0.0   \n",
       "...                          ...         ...         ...        ...      ...   \n",
       "SP16269                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP122676                     0.0         0.0         0.0        0.0      0.0   \n",
       "SP88776                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP64546                      0.0         0.0         0.0        0.0      0.0   \n",
       "SP21057                      0.0         0.0         0.0        0.0      0.0   \n",
       "\n",
       "                      Variant_Type  Variant_Classification  \n",
       "Tumor_Sample_Barcode                                        \n",
       "SP89389                        SNP       Nonsense_Mutation  \n",
       "SP21193                        SNP       Missense_Mutation  \n",
       "SP13206                        NaN                     NaN  \n",
       "SP103623                       NaN                     NaN  \n",
       "SP32742                        NaN                     NaN  \n",
       "...                            ...                     ...  \n",
       "SP16269                        NaN                     NaN  \n",
       "SP122676                       SNP       Nonsense_Mutation  \n",
       "SP88776                        NaN                     NaN  \n",
       "SP64546                        SNP       Missense_Mutation  \n",
       "SP21057                        DEL         Frame_Shift_Del  \n",
       "\n",
       "[1222 rows x 55923 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Preview the first few rows of the merged dataframe\n",
    "merged_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c997f49b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Data shape:(1222, 55923)\n"
     ]
    }
   ],
   "source": [
    "# Print the shape of the merged dataframe\n",
    "print(f\"Data shape:{merged_data.shape}\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "58819542",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Fill missing Variant_Classification values with 'No_Mutation'\n",
    "merged_data.Variant_Classification = merged_data.Variant_Classification.fillna(\"No_Mutation\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d3698b19",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Variant_Classification\n",
       "No_Mutation          803\n",
       "Missense_Mutation    264\n",
       "Nonsense_Mutation     50\n",
       "Frame_Shift_Del       40\n",
       "Splice_Site           33\n",
       "Frame_Shift_Ins       15\n",
       "In_Frame_Del           8\n",
       "In_Frame_Ins           6\n",
       "Splice_Region          2\n",
       "Silent                 1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Count the number of samples for each mutation classification\n",
    "merged_data.Variant_Classification.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c9cc39d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a copy of the merged dataframe for further processing\n",
    "merged_df = merged_data.copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "624deffe",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Class labels: {0: 'DEL', 1: 'DNP', 2: 'INS', 3: 'SNP', 4: 'nan'}\n"
     ]
    }
   ],
   "source": [
    "# --- Encode mutation type (multiclass target) ---\n",
    "le = LabelEncoder()\n",
    "merged_df['Variant_Type'] = le.fit_transform(merged_df['Variant_Type'].astype(str))  # Encode NaN as a class if needed\n",
    "\n",
    "# Save class label mapping\n",
    "class_names =  {0: 'nan', 1: 'SNV', 2: 'deletion', 3: 'insertion', 4: 'substitution'}\n",
    "print(\"Class labels:\", class_names)\n",
    "\n",
    "# --- Binary label: is TP53 mutated or not ---\n",
    "# 0 if Variant_Type was NaN, else 1\n",
    "merged_df['Mutated'] = np.where(merged_df['Variant_Type'] == le.transform(['nan'])[0], 0, 1)\n",
    "# Clean up column names by removing text in parentheses and stripping whitespace\n",
    "merged_df.columns = merged_df.columns.str.replace(r'\\(.*?\\)', '', regex=True).str.strip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d284635e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the processed merged dataframe to CSV\n",
    "merged_df.to_csv(f\"../data/processed/tcga/merged_tcga.csv\", index=True)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
